c++/src/capnp/rpc.c++

```diff
@@ -998,7 +993,9 @@ private:
     }
 
     kj::Maybe<kj::Promise<kj::Own<ClientHook>>> whenMoreResolved() override {
-      return fork.addBranch();
+      return fork.addBranch().then([this](auto&&) {
+        return cap->addRef();
```

Comment:
This seems to mean that if `resolve()` throws, then `whenMoreResolved()` will return the `initial` constructor parameter. The previous behavior was always to return the `eventual` constructor parameter. Is that intended, or am I reading the code wrong?

Reply:
`cap` is replaced during `resolve()`, so it's the same thing. But I think the reason I put this here was because I was thinking of making `resolve()` return void and then `fork` would be `ForkedPromise<void>`, but it looks like I didn't follow through on that... will fix.

Reply:
I guess it wouldn't make sense to return `eventual` if `resolve()` threw -- I would intuitively expect a broken capability, though.

Reply:
> cap is replaced during resolve(), so it's the same thing.

My point is that if `resolve()` throws, that's not necessarily true, since `cap` is replaced at the very end.

Reply:
At any rate, LGTM otherwise!

Reply:
If resolve() throws then this `.then()` will be skipped...

Reply:
By my reading, an exception in `resolve()` would be caught by the `.catch_()` on line 890, which would then pass `newBrokenCap()` to this `.then()`. I.e., it would not be skipped.

Reply:
Ah... yeah I don't think that's what I intended. Fixup: eb44b761